File: bench/cases/throw-catch-100k/main.ts
function trial(i: number): number {
  try {
    throw i
  } catch (e) {
    return e as number
  }
}

let total = 0
for (let i = 0; i < 100000; i = i + 1) {
  total = total + trial(i)
}
console.log(total)

File: bench/cases/throw-catch-100k/main.rs
use std::panic;

fn trial(i: i64) -> i64 {
    let result = panic::catch_unwind(panic::AssertUnwindSafe(|| -> i64 {
        panic::panic_any(i)
    }));
    *result.unwrap_err().downcast_ref::<i64>().unwrap()
}

fn main() {
    // Silence the default panic hook so the bench harness's stderr stays
    // clean — every trial intentionally panics.
    panic::set_hook(Box::new(|_| {}));
    let mut total: i64 = 0;
    for i in 0..100_000 {
        total = total + trial(i);
    }
    println!("{}", total);
}
